docs/dasharo-menu-docs: describe Dasharo menus (#419)
Dasharo/docs: docs/dasharo-menu-docs/overview.md

* [Boot Manager Menu](#boot-manager-menu) - entered with a different key than
  used for setup application. Lists all bootable options and allows one to
  override the boot path.

### Main Page

![](/images/menus/main_page.jpeg){ class="center" }

The page is the main view of the firmware setup application. It contains the
board model (`MS-7D25`), installed CPU and firmware version in the top-left
corner. In the top-right corner the CPU frequency and system RAM amount are
shown.

From the main page one may access all menus and submenus available in the
firmware setup. Besides the menus there is also an option to:

1. Change the language (currently only English is supported)
2. `Continue` - execute the top first boot order priority
3. `Reset` - resets the platform.

The currently available menus/submenus are as follows:

* [User Password Management](#user-password-management) - allows to set
  firmware setup password
* [Device Manager](device-manager.md) - allows configuring various devices and
  features like: UEFI Secure Boot, TPM device, SATA and TCG OPAL password, etc.
  It may also contain informational menus like Driver Health Manager, Network
  Device List and others.

> * [Dasharo System Features](dasharo-system-features.md)- contains submenus for

Reviewer: ```suggestion
* [Dasharo System Features](dasharo-system-features.md) - contains submenus for
```

  various UEFI standard variables responsible for console and boot options. One
  may choose which devices should be used for input and output, choose to boot
  an arbitrary file or modify the boot options and boot order.

NOTE: not all submenus may be available on your platform.
[Contact Dasharo Team](mailto:contact@dasharo.com) for more information and
possible feature extension of your platform.

### User Password Management

User Password Management menu allows one to set firmware setup admin password:

![](/images/menus/password_mgr.jpeg){ class=center" }

The password must be a strong one. The requirements are described on the help
section on the right side of the menu page. If the password is too weak it will
not be accepted. The page shows the current status of `Admin Password` if it
was provisioned or not.

To disable the password, enter the `User Password Management` menu again and
request to set new password. You will be asked to enter current password. When
a window to provide new password pops up, leave the field empty and confirm it
twice with `ENTER`. The password installation status on the menu page should

> change to `Not Installed`.

Reviewer: It would be good to add information, that if anyone would like to set a new password, it must be different from the last 5 passwords.

Author: @sulewskiprzemyslaw is this a recommendation or just how the feature works?

Reviewer: @miczyg1 feature works in that way

Author: Thanks, added that information

### One Time Boot

When setup application is entered, one of the menus is called `One Time Boot`.
As the name suggests it allows to override the boot just one time (not
permanently). The usage principles are the same as for [Boot Manager Menu](#boot-manager-menu).

Example view of `One Time Boot` submenu:

![](/images/menus/one_time_boot.jpeg){ class="center" }

On the right side of the menu window, there is a `DevicePath` which is a
UEFI-compliant path to the device or file being executed. Depending on the
file/device type, these paths may be different:

![](/images/menus/one_time_boot.jpeg){ class="center" }

![](/images/menus/one_time_boot3.jpeg){ class="center" }

`DevicePaths` are defined in [UEFI Specification](https://uefi.org/specifications).
If you are interested in decoding those, read through the specification
carefully.

### Boot Manager Menu

Boot Manager Menu is an application that lists all bootable options and allows
one to override the boot path.

Boot Manager Menu is entered with a different key than setup application. It
may be customized on your platform. The right key to use is always printed on
the screen in the top-left corner, for example `F11`:

> ```txt

Reviewer: Information about entering the menus is also available at the beginning of the documentation. Perhaps a better solution is to indicate access to the menus at the beginning of the document (with additional information or a link to dedicated documentation which will contain a labeled list of keys for all Dasharo products)

Author: Yes, but it is just mentioned briefly o the beginning. Here is more detailed description

## Dasharo menu guides

> The main menu is entered by selecting the Setup application in the

Reviewer 2: I would drop the `[Here](/variants/novacustom_nv4x_tgl/compatibility-check-results-ubuntu.md) you
can see checkbox results on Dasharo firmware.` a few lines above while editing this file. It does not fit there, at all.

Author: I don't understand. Where do you see `[Here](/variants/novacustom_nv4x_tgl/compatibility-check-results-ubuntu.md)` in this file?

Reviewer 2: Few lines above, in the original file (not your changes).

I just reviewed the thing as a whole, and the checbox stuff struck me as it did not fit there at all.
I understand that this is not what you added or modified here, but I would remove it or move somehtere esle when editing this overview.md file.

Reviewer 2: There is also a link to some video, you may judge how useful it is nowadays in scepe of the current menu layout.

Author: Ok I see. Totally agree.

Author: Removed.

> The main menu is entered by executing Setup application either through
> [Boot Manager Menu](#boot-manager-menu) or platform-specific hotkey. You may
> find the right key in the top-left corner when booting the platform (when logo
> shows up), for example `DEL`:

Reviewer 3: ```suggestion
The main menu is entered by selecting the Setup application in the
[Boot Manager Menu](#boot-manager-menu) or by pressing the Setup hotkey
while booting. The hotkey is shown in the top-left corner of the screen when
the boot logo is displayed, for example `DEL`:
```

```txt
DEL   to enter Setup
F11   to enter Boot Manager Menu
ENTER to boot directly
```

* [Setup Main Page](#main-page)
    - [User Password Management](#user-password-management)
    - [Device Manager](device-manager.md)
    - [Dasharo System Features](dasharo-system-features.md)
    - [One Time Boot](#one-time-boot)
    - [Boot Maintenance Manager](boot-maintenance-mgr.md)

* [Boot Manager Menu](#boot-manager-menu) - entered with a different key than
  used for setup application. Lists all bootable options and allows one to
  override the boot path.

### Main Page

![](/images/menus/main_page.jpeg){ class="center" }

The page is the main view of the firmware setup application. It contains the
board model (`MS-7D25`), installed CPU and firmware version in the top-left

> corner. In the top-right corner the CPU frequency and system RAM amount is

Reviewer 3: ```suggestion
corner. In the top-right corner the CPU frequency and system RAM amount are
```

  features specific to Dasharo products and Dasharo supported platforms
* [One Time Boot](#one-time-boot) - allows to choose which boot entry to
  execute. It simply lists all available boot options and allows to select one
  the same way as [Boot Manager Menu](#boot-manager-menu)
* [Boot Maintenance Manager](boot-maintenance-mgr.md) - allows to manipulate
  various UEFI standard variables responsible for console and boot options. One
  may choose which devices should be used for input and output, choose to boot
  an arbitrary file or modify the boot options and boot order.

NOTE: not all submenus may be available on your platform.
[Contact Dasharo Team](mailto:contact@dasharo.com) for more information and
possible feature extension of your platform.

### User Password Management

User Password Management menu allows one to set firmware setup admin password:

![](/images/menus/password_mgr.jpeg){ class=center" }

The password must be a strong one. The requirements are described on the help
section on the right side of the menu page. If the password is too weak it will
not be accepted. The page shows the current status of `Admin Password` if it
was provisioned or not.

> TO disable the password, enter the `User Password Management` menu again and

Reviewer 3: ```suggestion
To disable the password, enter the `User Password Management` menu again and
```

request to set new password. You will be asked to enter current password. When

> a windows to provide new password pops up, leave the field empty and confirm it

Reviewer 3: ```suggestion
a window to provide new password pops up, leave the field empty and confirm it
```

Example view of `One Time Boot` submenu:

![](/images/menus/one_time_boot.jpeg){ class="center" }

On the right side of the menu window, there is a `DevicePath` which is a
UEFI-compliant path to the device or file being executed. Depending on the
file/device type, these paths may be different:

![](/images/menus/one_time_boot.jpeg){ class="center" }

![](/images/menus/one_time_boot3.jpeg){ class="center" }

`DevicePaths` are defined in [UEFI Specification](https://uefi.org/specifications).
If you are interested in decoding those, read through the specification
carefully.

### Boot Manager Menu

Boot Manager Menu is an application that lists all bootable options and allows
one to override the boot path.

Boot Manager Menu is entered with a different key than setup application. It
may be customized on your platform. The right key to use is always printed on
the screen in the top-left corner, for example `F11`:

```txt
DEL   to enter Setup
F11   to enter Boot Manager Menu
ENTER to boot directly
```

After pressing the right hotkey for Boot Manager Menu, a window should pop up:

![](/images/menus/boot_manager.jpeg){ class="center" }

> On the bottom of the windows there are instruction how to move and select

Reviewer 3: ```suggestion
On the bottom of the window there are instruction how to move and select
```

  execute. It simply lists all available boot options and allows to select one
  the same way as [Boot Manager Menu](#boot-manager-menu)
* [Boot Maintenance Manager](boot-maintenance-mgr.md) - allows to manipulate
  various UEFI standard variables responsible for console and boot options. One
  may choose which devices should be used for input and output, choose to boot
  an arbitrary file or modify the boot options and boot order.

NOTE: not all submenus may be available on your platform.
[Contact Dasharo Team](mailto:contact@dasharo.com) for more information and
possible feature extension of your platform.

### Boot Manager Menu

Boot Manager Menu is an application that lists all bootable options and allows
one to override the boot path.

Boot Manager Menu is entered with a different key than setup application. It
may be customized on your platform. The right key to use is always printed on
the screen in the top-left corner, for example `F11`:

```txt
DEL   to enter Setup
F11   to enter Boot Manager Menu
ENTER to boot directly
```

After pressing the right hotkey for Boot Manager Menu, a window should pop up:

![](/images/menus/boot_manager.jpeg)

> One may use up and down arrows to move to desired position. Pressing enter will

Reviewer 2: Instead of explaining how to move in specific section, we could add one dedicated section for navigation, like here: https://www.dell.com/support/manuals/pl-pl/precision-15-7520-laptop/prec7520_om_pub/navigation-keys?guid=guid-f448dd64-1b07-4857-a446-42e313a39cc8&lang=en-us

Author: Navigation is also explaiend on the picture. So maybe I am overrepeating myself...

Author: Removed it.
